docs/tutorial_hist.rst

```diff
@@ -0,0 +1,402 @@
+how you display the path to this file. For example, 
+
+.. code-block:: xonshcon
+
+    >>> history file
+    /home/me/.local/share/xonsh/xonsh-ace97177-f8dd-4a8d-8a91-a98ffd0b3d17.json
+
+Note that by these files are stored in ``$XONSH_DATA_DIR`` environment variable. This 
+is, by default, set to the ``xonsh`` dir inside of the free desktop standards 
+``$XDG_DATA_HOME`` environment variable. See 
+`this page <http://standards.freedesktop.org/basedir-spec/latest/ar01s03.html>`_ for
+more details.
+
+``info`` action
+===============
+The info action combines the ``id`` and ``file`` actions as well as adds some additional
+information about the current state of the history. By default, this prints a key-value
+series of lines. However, it can also return a JSON formatted string.
+
+.. code-block:: xonshcon
+
+    >>> history info
+    sessionid: ace97177-f8dd-4a8d-8a91-a98ffd0b3d17
+    filename: /home/scopatz/.local/share/xonsh/xonsh-ace97177-f8dd-4a8d-8a91-a98ffd0b3d17.json
+    length: 6
+    buffersize: 100
+    bufferlength: 6
+
+.. code-block:: xonshcon
+
+    >>> history info --json
+    {"sessionid": "ace97177-f8dd-4a8d-8a91-a98ffd0b3d17", 
+     "filename": "/home/scopatz/.local/share/xonsh/xonsh-ace97177-f8dd-4a8d-8a91-a98ffd0b3d17.json", 
+     "length": 7, "buffersize": 100, "bufferlength": 7}
+
+``replay`` action
+==================
+The ``replay`` action allows for history files to be rerun, as scripts or in an existing xonsh 
+session.
+
+First, the original ``'replay'`` environment is loaded and will be merged with the current ``'native'`` 
+environment. How the environments are merged or not merged can be set at replay time. The default is for 
+the current native environment to take precedence. Next, each input in the environment is executed in order. 
+Lastly, the information of the replayed history file is printed.
+
+Let's walk through an example. To begin with, open up xonsh and run some simple commands, as follows.
+Call this the ``orig`` session.
+
+**orig history**
+
+.. code-block:: xonshcon
+
+    >>> mkdir -p temp/
+    >>> cd temp
+    >>> import random
+    >>> touch @(random.randint(0, 18))
+    >>> ls
+    2
+    >>> history file
+    /home/scopatz/.local/share/xonsh/xonsh-4bc4ecd6-3eba-4f3a-b396-a229ba2b4810.json
+    >>> exit
+
+We can now replay this by passing the filename into the replay command or the replay action 
+of the history command. This action has a few different options, but one of them is that 
+we can select a different target output file with the ``-o`` or ``--target`` option. 
+For example, in a new session, we could run:
+
+**new history**
+
+.. code-block:: xonshcon
+
+    >>> history replay -o ~/new.json ~/.local/share/xonsh/xonsh-4bc4ecd6-3eba-4f3a-b396-a229ba2b4810.json
+    2  10
+    /home/scopatz/new.json
+
+    ------------------------------------------------------------
+    Just replayed history, new history has following information
```

Comment:
Thanks, got it.

On Thu, Aug 27, 2015 at 5:16 PM Morten Enemark Lund <
notifications@github.com> wrote:

> In docs/tutorial_hist.rst
> <https://github.com/scopatz/xonsh/pull/353#discussion_r38147141>:
>
> > +
> > +We can now replay this by passing the filename into the replay command or the replay action
> > +of the history command. This action has a few different options, but one of them is that
> > +we can select a different target output file with the ``-o`` or ``--target`` option.
> > +For example, in a new session, we could run:
> > +
> > +**new history**
> > +
> > +.. code-block:: xonshcon
> > +
> > +    >>> history replay -o ~/new.json ~/.local/share/xonsh/xonsh-4bc4ecd6-3eba-4f3a-b396-a229ba2b4810.json
> > +    2  10
> > +    /home/scopatz/new.json
> > +
> > +    ------------------------------------------------------------
> > +    Just replayed history, new history has following information
>
> ... has *the* following ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scopatz/xonsh/pull/353/files#r38147141>.
>